dvc/commands/repro.py

```diff
@@ -118,8 +118,8 @@ def add_arguments(repro_parser):
         action="store_true",
         default=False,
         help=(
-            "Try automatically pulling missing cache for outputs restored "
-            "from the run-cache."
+            "Try automatically pulling missing dependencies and outputs. "
+            "Implies --allow-missing."
```

Comment:
would it better to deprecate then allow-missing? and here describe the behavior in a more explicit way (by not referring to another option, but actually saying what is happening)

Reply:
I didn't want to deprecate now because:
1. That is a clear breaking change that I don't think we should do unless we want to do a major version change
2. Unlike `--pull`, `--allow-missing` on its own can be useful. It was introduced to solve https://github.com/iterative/dvc/issues/5369, in which case the user doesn't want to pull anything.

Open to other ideas on how to explain it but worried about overcomplicating it. I think this is how people expect `--pull` works now. With `--pull` and `--allow-missing`, if you have run the stage before and pushed it, it won't get run again. Without `--allow-missing`, the stage may still be considered changed and run again even if it can be pulled from the run-cache.